types/http-errors/index.d.ts

```diff
@@ -22,16 +22,21 @@ declare namespace createHttpError {
         [key: string]: any;
     }
 
-    type UnknownError = Error | string | number | { [key: string]: any };
+    type UnknownError = Error | string | { [key: string]: any };
 
-    type HttpErrorConstructor<N extends number = number> = new (msg?: string) => HttpError<N>;
+    interface HttpErrorConstructor<N extends number = number> {
+        (msg?: string): HttpError<N>;
+        new (msg?: string): HttpError<N>;
+    }
 
-    type CreateHttpError = <N extends UnknownError>(arg: N, ...rest: UnknownError[]) => HttpError<N extends number ? N : number>;
+    interface CreateHttpError {
+        <N extends number = number>(arg?: N, ...rest: UnknownError[]): HttpError<N>;
```

Comment:
Does `arg` really need to be optional here? The second signature should cover the case that no error code is provided. Otherwise, a call like this would be allowed and I'm not sure whether it's valid, I think it will throw:

```ts
create(undefined, 'Not Found');
```

Reply:
You're right, I made it non-optional now, thanks.